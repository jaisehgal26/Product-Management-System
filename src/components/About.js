import React, { Suspense } from "react";
import about_1 from "../img/about-1.jpg";
import about_2 from "../img/about-2.jpg";

import Fade from "react-reveal/Fade";
import Footer from "./Footer";
import Navbar from "./Navbar";

const Service = React.lazy(() => import("./Service"));
const ChefTeam = React.lazy(() => import("./ChefTeam"));

export default function About() {
  return (
    <>
      <Navbar />

      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <div className="container-xxl py-6">
          <div className="container">
            <div className="row g-5">
              <Fade bottom>
                <div className="col-lg-6 animate__animated animate__fadeInUp">
                  <div className="row img-twice position-relative h-100">
                    <div className="col-6">
                      <img
                        className="img-fluid rounded"
                        src={about_1}
                        alt=""
                        loading="lazy"
                      />
                    </div>
                    <div className="col-6 align-self-end">
                      <img
                        className="img-fluid rounded"
                        src={about_2}
                        alt=""
                        loading="lazy"
                      />
                    </div>
                  </div>
                </div>
              </Fade>
              <Fade bottom>
                <div className="col-lg-6 animate__animated animate__fadeInUp">
                  <div className="h-100">
                    <p className="text-primary text-uppercase mb-2">
                      # About Us
                    </p>
                    <h1 className="display-6 mb-4 Playfair">
                      We Bake Every Item From The Core Of Our Hearts
                    </h1>
                    <p className="Roboto">
                      Tempor erat elitr rebum at clita. Diam dolor diam ipsum
                      sit. Aliqu diam amet diam et eos. Clita erat ipsum et
                      lorem et sit, sed stet lorem sit clita duo justo magna
                      dolore erat amet
                    </p>
                    <p>
                      Tempor erat elitr rebum at clita. Diam dolor diam ipsum
                      sit. Aliqu diam amet diam et eos. Clita erat ipsum et
                      lorem et sit, sed stet lorem sit clita duo justo magna
                      dolore erat amet
                    </p>
                    <div className="row g-2 mb-4">
                      <div className="col-sm-6">
                        <i className="fa fa-check text-primary me-2"></i>Quality
                        Products
                      </div>
                      <div className="col-sm-6">
                        <i className="fa fa-check text-primary me-2"></i>Custom
                        Products
                      </div>
                      <div className="col-sm-6">
                        <i className="fa fa-check text-primary me-2"></i>Online
                        Order
                      </div>
                      <div className="col-sm-6">
                        <i className="fa fa-check text-primary me-2"></i>Home
                        Delivery
                      </div>
                    </div>
                  </div>
                </div>
              </Fade>
            </div>
          </div>
        </div>
        <ChefTeam />

        <Service />
      </Suspense>
      <Footer />
    </>
  );
}
