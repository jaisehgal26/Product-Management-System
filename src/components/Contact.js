import React, { Suspense } from "react";
import { Formik, Form, Field, ErrorMessage, useFormikContext } from "formik";
import * as Yup from "yup";
import Axios from "axios";
import Navbar from "./Navbar";
import { Prompt } from "react-router-dom";
import Footer from "./Footer";
import Fade from "react-reveal/Fade";

export default function Contact() {
  const initialValues = { name: "", email: "", subject: "", message: "" };
  const onSubmit = (value, { resetForm }) => {
    Axios.post("http://localhost:3000/contact", {
      name: value.name,
      email: value.email,
      subject: value.subject,
      message: value.message,
    });
    resetForm({ value: "" });
  };
  const PromptIfDirty = () => {
    const formik = useFormikContext();
    return (
      <Prompt
        when={formik.dirty && formik.submitCount === 0}
        message="Are you sure you want to leave? You have with unsaved changes."
      />
    );
  };
  const validationSchema = Yup.object({
    name: Yup.string().required("Required"),
    email: Yup.string().email("Enter valid Email").required("Required"),
    subject: Yup.string().required("Required"),
    message: Yup.string().required("Required"),
  });
  return (
    <>
      <Navbar />

      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <Fade bottom>
          <div className="container-xxl py-6">
            <div className="container">
              <div
                className="text-center mx-auto mb-5 "
                style={{ maxwidth: "500px" }}
              >
                <p className="text-primary text-uppercase mb-2"># Contact Us</p>
                <h1 className="display-6 mb-4 Playfair">
                  If You Have Any Query, Please Contact Us
                </h1>
              </div>
              <div className="row g-0 justify-content-center">
                <div className="col-lg-8 ">
                  <Formik
                    initialValues={initialValues}
                    validationSchema={validationSchema}
                    onSubmit={onSubmit}
                  >
                    <Form>
                      <PromptIfDirty />
                      <div className="row g-3">
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="name"
                              name="name"
                              className="form-control"
                            />
                            <label htmlFor="name" className="form-label">
                              Name
                            </label>
                            <ErrorMessage name="name" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="email"
                              name="email"
                              className="form-control"
                            />
                            <label htmlFor="email" className="form-label">
                              Email
                            </label>
                            <ErrorMessage name="email" />
                          </div>
                        </div>
                        <div className="col-12">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="subject"
                              name="subject"
                              className="form-control"
                            />
                            <label htmlFor="subject" className="form-label">
                              Subject
                            </label>
                            <ErrorMessage name="subject" />
                          </div>
                        </div>
                        <div className="col-12">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="message"
                              name="message"
                              className="form-control"
                              style={{ height: "5rem" }}
                            />
                            <label htmlFor="message" className="form-label">
                              Message
                            </label>
                            <ErrorMessage name="message" />
                          </div>
                        </div>
                        <div className="col-12 text-center">
                          <button
                            className="btn btn-primary rounded-pill py-3 px-5"
                            type="submit"
                          >
                            Send Message
                          </button>
                        </div>
                      </div>
                    </Form>
                  </Formik>
                </div>
              </div>
            </div>
          </div>
        </Fade>
      </Suspense>
      <Footer />
    </>
  );
}
