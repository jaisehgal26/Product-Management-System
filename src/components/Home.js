import React, { Suspense } from "react";
import Carousel from "./Carousel";
import Footer from "./Footer";
import Navbar from "./Navbar";

const Service = React.lazy(() => import("./Service"));
const Review = React.lazy(() => import("./Review"));
const Facts = React.lazy(() => import("./Facts"));
const Banner = React.lazy(() => import("./Banner"));
export default function Home() {
  return (
    <>
      <Navbar />
      <Carousel />
      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <Facts />
        <Service />
        <Banner />
        <Review />
      </Suspense>
      <Footer />
    </>
  );
}
