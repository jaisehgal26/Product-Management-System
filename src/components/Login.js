import React, { Suspense, useState } from "react";
import { Formik, Form, Field, ErrorMessage, useFormikContext } from "formik";
import * as Yup from "yup";
import Axios from "axios";
import { Link, useHistory, Prompt } from "react-router-dom";
import Fade from "react-reveal/Fade";

import Footer from "./Footer";
import Navbar from "./Navbar";
import { useAuth } from "./contexts/AuthContext";

export default function Login() {
  const { login } = useAuth();
  const [error, setError] = useState("");
  const [loading, setLoading] = React.useState(false);
  const history = useHistory();
  const initialValues = {
    email: "",
    password: "",
  };
  const PromptIfDirty = () => {
    const formik = useFormikContext();
    return (
      <Prompt
        when={formik.dirty && formik.submitCount === 0}
        message="Are you sure you want to leave..?"
      />
    );
  };
  const onSubmit = async (value, { resetForm }) => {
    try {
      setError("");
      setLoading(true);
      await login(value.email, value.password);
      Axios.post("http://localhost:3000/login", {
        email: value.email,
        password: value.password,
      });
      history.push("/");
    } catch {
      setError("Failed to Sign In Email or Password is Incorrect");
    }
    setLoading(false);
    resetForm({ value: "" });
  };
  const validationSchema = Yup.object({
    email: Yup.string().email("Enter valid Email").required("Required"),
    password: Yup.string().required("Required"),
  });
  return (
    <>
      <Navbar />

      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <Fade bottom>
          <div className="container-xxl py-6">
            <div className="container">
              <div
                className="text-center mx-auto mb-5 "
                style={{ maxwidth: "500px" }}
              >
                <p className="text-primary text-uppercase mb-2"># Login</p>
                <h1 className="display-6 mb-4 Playfair">
                  Login here to access other features
                </h1>
                {error && (
                  <div className="alert-danger" role="alert">
                    {error}
                  </div>
                )}
              </div>

              <div className="row g-0 justify-content-center">
                <div className="col-lg-8 ">
                  <Formik
                    initialValues={initialValues}
                    validationSchema={validationSchema}
                    onSubmit={onSubmit}
                  >
                    <Form>
                      <PromptIfDirty />
                      <div className="row g-3">
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="email"
                              id="email"
                              name="email"
                              className="form-control"
                            />
                            <label htmlFor="email" className="form-label">
                              Email
                            </label>
                            <ErrorMessage name="email" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="password"
                              id="password"
                              name="password"
                              className="form-control"
                            />
                            <label htmlFor="password" className="form-label">
                              Password
                            </label>
                            <ErrorMessage name="password" />
                          </div>
                        </div>
                        <div className="col-12 text-center">
                          <button
                            className="btn btn-primary rounded-pill py-3 px-5 mx-3"
                            type="submit"
                            disabled={loading}
                          >
                            Login
                          </button>
                        </div>
                        <div className="col-12 text-center">
                          <Link to="/forgotpassword" className=" py-3 px-2 ">
                            Forgot Password?
                          </Link>
                        </div>
                        <div className="col-12 text-center">
                          Need an Account -
                          <Link to="/signup" className=" py-3 px-2 ">
                            Create Account
                          </Link>
                        </div>
                      </div>
                    </Form>
                  </Formik>
                </div>
              </div>
            </div>
          </div>
        </Fade>
      </Suspense>
      <Footer />
    </>
  );
}
