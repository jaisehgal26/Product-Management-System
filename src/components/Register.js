import React, { Suspense, useState } from "react";
import { Formik, Form, Field, ErrorMessage, useFormikContext } from "formik";
import * as Yup from "yup";
import Axios from "axios";
import { Link, useHistory, Prompt } from "react-router-dom";
import Navbar from "./Navbar";
import { useAuth } from "./contexts/AuthContext";

import Fade from "react-reveal/Fade";
import Footer from "./Footer";

export default function Register() {
  const { signup } = useAuth();
  const [error, setError] = useState("");
  const [loading, setLoading] = useState(false);
  const history = useHistory();
  const initialValues = {
    fname: "",
    lname: "",
    email: "",
    password: "",
    confirm_password: "",
    phone_number: "",
    location: "",
    product1: "",
    product2: "",
    product3: "",
    product4: "",
    vid: "",
  };
  const PromptIfDirty = () => {
    const formik = useFormikContext();
    return (
      <Prompt
        when={formik.dirty && formik.submitCount === 0}
        message="Are you sure you want to leave? You have with unsaved changes."
      />
    );
  };
  const onSubmit = async (value, { resetForm }) => {
    if (value.password !== value.confirm_password) {
      return setError("Passwords do not match");
    }

    try {
      setError("");
      setLoading(true);
      await signup(value.email, value.password);
      Axios.post("http://localhost:3000/register", {
        fname: value.fname,
        lname: value.lname,
        email: value.email,
        password: value.password,
        confirm_password: value.confirm_password,
        phone_number: value.phone_number,
        location: value.location,
        product1: value.product1,
        product2: value.product2,
        product3: value.product3,
        product4: value.product4,
        vid: value.vid,
      });
      history.push("/");
    } catch {
      setError("Failed to create an account");
    }
    setLoading(false);
    resetForm({ value: "" });
  };
  const validationSchema = Yup.object({
    fname: Yup.string().required("Required"),
    lname: Yup.string().required("Required"),
    email: Yup.string().email("Enter valid Email").required("Required"),
    password: Yup.string().required("Required"),
    confirm_password: Yup.string().required("Required"),
    phone_number: Yup.string().required("Required"),
    location: Yup.string().required("Required"),
    product1: Yup.string().required("Required"),
    product2: Yup.string().required("Required"),
    product3: Yup.string().required("Required"),
    product4: Yup.string().required("Required"),
    vid: Yup.string().required("Required"),
  });

  return (
    <>
      <Navbar />

      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <Fade bottom>
          <div className="container-xxl py-6">
            <div className="container">
              <div
                className="text-center mx-auto mb-5 "
                style={{ maxwidth: "500px" }}
              >
                <p className="text-primary text-uppercase mb-2"># Register</p>

                <h1 className="display-6 mb-4">Register for new User</h1>
                {error && (
                  <div className="alert-danger" role="alert">
                    {error}
                  </div>
                )}
              </div>
              <div className="row g-0 justify-content-center">
                <div className="col-lg-8 ">
                  <Formik
                    initialValues={initialValues}
                    validationSchema={validationSchema}
                    onSubmit={onSubmit}
                  >
                    <Form>
                      <div className="row g-3">
                        <div className="col-md-6">
                          <div className="form-floating">
                            <PromptIfDirty />
                            <Field
                              type="text"
                              id="fname"
                              name="fname"
                              className="form-control"
                            />
                            <label htmlFor="fname" className="form-label">
                              First Name
                            </label>
                            <ErrorMessage name="fname" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="lname"
                              name="lname"
                              className="form-control"
                            />
                            <label htmlFor="lname" className="form-label">
                              Last Name
                            </label>
                            <ErrorMessage name="lname" />
                          </div>
                        </div>
                        <div className="col-6">
                          <div className="form-floating">
                            <Field
                              type="email"
                              id="email"
                              name="email"
                              className="form-control"
                            />
                            <label htmlFor="email" className="form-label">
                              Email
                            </label>
                            <ErrorMessage name="email" />
                          </div>
                        </div>
                        <div className="col-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="vid"
                              name="vid"
                              className="form-control"
                            />
                            <label htmlFor="vid" className="form-label">
                              Vender ID
                            </label>
                            <ErrorMessage name="vid" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="password"
                              id="password"
                              name="password"
                              className="form-control"
                            />
                            <label htmlFor="password" className="form-label">
                              Password
                            </label>
                            <ErrorMessage name="password" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="password"
                              id="confirm_password"
                              name="confirm_password"
                              className="form-control"
                            />
                            <label
                              htmlFor="confirm_password"
                              className="form-label"
                            >
                              Confirm Password
                            </label>
                            <ErrorMessage name="confirm_password" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="number"
                              id="phone_number"
                              name="phone_number"
                              className="form-control"
                              maxLength="10"
                            />
                            <label
                              htmlFor="phone_number"
                              className="form-label"
                            >
                              Phone Number
                            </label>
                            <ErrorMessage name="phone_number" />
                          </div>
                        </div>
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="location"
                              id="location"
                              name="location"
                              className="form-control"
                            />
                            <label htmlFor="location" className="form-label">
                              location
                            </label>
                            <ErrorMessage name="location" />
                          </div>
                        </div>{" "}
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="product1"
                              name="product1"
                              className="form-control"
                            />
                            <label htmlFor="product1" className="form-label">
                              Product
                            </label>
                            <ErrorMessage name="product1" />
                          </div>
                        </div>{" "}
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="product2"
                              name="product2"
                              className="form-control"
                            />
                            <label htmlFor="product2" className="form-label">
                              Product
                            </label>
                            <ErrorMessage name="product2" />
                          </div>
                        </div>{" "}
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="product3"
                              name="product3"
                              className="form-control"
                            />
                            <label htmlFor="product3" className="form-label">
                              Product
                            </label>
                            <ErrorMessage name="product3" />
                          </div>
                        </div>{" "}
                        <div className="col-md-6">
                          <div className="form-floating">
                            <Field
                              type="text"
                              id="product4"
                              name="product4"
                              className="form-control"
                            />
                            <label htmlFor="product4" className="form-label">
                              Product
                            </label>
                            <ErrorMessage name="product4" />
                          </div>
                        </div>
                        <div className="col-12 text-center">
                          <button
                            className="btn btn-primary rounded-pill py-3 px-5"
                            type="submit"
                            disabled={loading}
                          >
                            Sign Up
                          </button>
                        </div>
                        <div className="col-12 text-center">
                          Already have an account -
                          <Link to="/login" className=" py-3 px-2">
                            Login
                          </Link>
                        </div>
                      </div>
                    </Form>
                  </Formik>
                </div>
              </div>
            </div>
          </div>
        </Fade>
      </Suspense>

      <Footer />
    </>
  );
}
