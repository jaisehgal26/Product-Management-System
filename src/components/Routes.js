import React, { Suspense, lazy } from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import { AuthProvider } from "./contexts/AuthContext";
import Home from "./Home";
import About from "./About";
import Contact from "./Contact";
import Products from "./Product/AllProductsPage";
import Register from "./Register";
import Login from "./Login";
import "../App.css";
const ProductDetail = lazy(() => import("./Product/ProductDetail"));
const AddProduct = lazy(() => import("./Product/AddProduct"));
const ForgotPassword = lazy(() => import("./ForgotPassword"));
const ProfileDetails = lazy(() => import("./ProfileDetail"));
const UpdateProfile = lazy(() => import("./UpdateProfile"));
const EditProduct = lazy(() => import("./Product/EditProduct"));
const ContactMessage = lazy(() => import("./ContactMessage"));

export default function Routes() {
  return (
    <Router>
      <Suspense
        fallback={
          <div className="center spinner-grow text-warning" role="status">
            <span className="sr-only">Loading...</span>
          </div>
        }
      >
        <AuthProvider>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route
              exact
              path="/products/addproduct"
              component={AddProduct}
            ></Route>
            <Route
              exact
              path="/products/editproduct/:id"
              component={EditProduct}
            ></Route>
            <Route exact path="/products/:id" component={ProductDetail}></Route>

            <Route path="/about" component={About}></Route>

            <Route path="/contactmessages" component={ContactMessage}></Route>

            <Route path="/profile" component={ProfileDetails}></Route>
            <Route path="/forgotpassword" component={ForgotPassword}></Route>
            <Route path="/update-profile" component={UpdateProfile}></Route>
            <Route path="/signup" component={Register}></Route>
            <Route path="/login" component={Login}></Route>
            <Route path="/products" component={Products}></Route>
            <Route path="/contact" component={Contact}></Route>
          </Switch>
        </AuthProvider>
      </Suspense>
    </Router>
  );
}
