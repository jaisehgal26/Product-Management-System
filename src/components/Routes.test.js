import React from "react";
import { Route } from "react-router";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Routes from "./Routes";
import Home from "./Home";
import About from "./About";
import Products from "./Product/AllProductsPage";
import Contact from "./Contact";
import Register from "./Register";
import Login from "./Login";

test("should render correctly", () => {
  const wrapper = toJson(shallow(<Routes />));
  expect(wrapper).toMatchSnapshot();
});

test("renders correct route home", () => {
  const wrapper = shallow(<Routes />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap["/"]).toBe(Home);
});
test("renders correct route about", () => {
  const wrapper = shallow(<Routes />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap["/about"]).toBe(About);
});
test("renders correct route products", () => {
  const wrapper = shallow(<Routes />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap["/products"]).toBe(Products);
});
test("renders correct route contact", () => {
  const wrapper = shallow(<Routes />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap["/contact"]).toBe(Contact);
});
test("renders correct route register", () => {
  const wrapper = shallow(<Routes />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap["/signup"]).toBe(Register);
});
test("renders correct route login", () => {
  const wrapper = shallow(<Routes />);
  const pathMap = wrapper.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});

  expect(pathMap["/login"]).toBe(Login);
});
