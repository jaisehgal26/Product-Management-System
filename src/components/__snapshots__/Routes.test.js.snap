// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<BrowserRouter>
  <Suspense
    fallback={
      <div
        className="center spinner-grow text-warning"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading...
        </span>
      </div>
    }
  >
    <AuthProvider>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          exact={true}
          path="/products/addproduct"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          exact={true}
          path="/products/editproduct/:id"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          exact={true}
          path="/products/:id"
        />
        <Route
          component={[Function]}
          path="/about"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          path="/contactmessages"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          path="/profile"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          path="/forgotpassword"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_init": [Function],
              "_payload": Object {
                "_result": [Function],
                "_status": -1,
              },
            }
          }
          path="/update-profile"
        />
        <Route
          component={[Function]}
          path="/signup"
        />
        <Route
          component={[Function]}
          path="/login"
        />
        <Route
          component={[Function]}
          path="/products"
        />
        <Route
          component={[Function]}
          path="/contact"
        />
      </Switch>
    </AuthProvider>
  </Suspense>
</BrowserRouter>
`;
